Docker.md

```diff
@@ -0,0 +1,171 @@
+## Docker
+
+[Docker](https://www.docker.com) is a tool that enables packaging software as a [container image](https://www.docker.com/resources/what-container) that includes all of the dependencies to run a given application. In the context of this project, Docker enables building and running the `tbb` application across multiple operating systems.
+
+### Installation
+
+To get started, follow the official [Getting Started](https://docs.docker.com/get-docker/) guide to install Docker for your operating system. If you are using Linux, be sure to also follow [the guide for installing docker-compose](https://docs.docker.com/compose/install/#install-compose-on-linux-systems). This extra step is not necessary on Windows and MacOS.
+
+Once all tools are installed, you will have two new executables that you can run in your terminal:
+1. `docker` - This is the application used for building and running containers
+2. `docker-compose` - [Docker Compose](https://docs.docker.com/compose/) is a tool for defining how to run any number of containers
+
+You can check that both of these applications are installed by running the following commands in your terminal:
+
+```
+$ which docker
+/usr/local/bin/docker
+
+$ which docker-compose
+/usr/local/bin/docker-compose
+```
+
+This uses the UNIX [which](https://www.tutorialspoint.com/unix_commands/which.htm) command to print the absolute path to the executable in question. As long as a path is printed for both `docker` and `docker-compose` then you are good to go - the paths do not need to be the same as those shown above.
+
+### Building
+
+You can build the Docker image for the `tbb` application by running `make image` in the root of this repository. This will take a minute or two to build and the output in your terminal will look something like this:
+
+```
+$ make image
```

Comment:
Nice, even with a Makefile command :100: 

I will add this to the main repo's README.md -> Usage -> Install section, another section called `Install with Docker` with instructions for your docker-compose.